generator/generator_config.textproto

```diff
@@ -382,6 +388,31 @@ service {
   initial_copyright_year: "2022"
   retryable_status_codes: ["kUnavailable"]
   emitted_rpcs: ["SetLocations"]
+  omitted_rpcs: [
+    "CreateCluster(std::string const&, std::string const&, google::container::v1::Cluster const&)",
+    "UpdateCluster(std::string const&, std::string const&, std::string const&, google::container::v1::ClusterUpdate const&)",
+    "GetCluster(std::string const&, std::string const&, std::string const&)",
+    "ListClusters(std::string const&, std::string const&)",
```

Comment:
Let's assume we have never generated the method because it has always contained a deprecated field.

The service team could introduce a `(std::string const&, std::string const&)` method signature that uses a different, non-deprecated field. We would want to generate that new method. But we would omit it, because the method signature is listed here.

I think we have to use the method signature index in some way to differentiate between these overload sets.

Reply:
Hmmm... you mean `FooMethod(std::string const& field1)` and `FooMethod(std::string const& field2)`?  That is ambiguous and not implementable anyway, right?

Reply:
If `field1` is deprecated, but `field2` is not, we should generate a `FooMethod(std::string const& field2)`. (currently, we would not generate either).

> That is ambiguous and not implementable anyway, right?

I think it can be implemented. The method signatures are ordered. The breaking change detector ensures that order is fixed (thanks to @tmatsuo).

It wouldn't work if a method signature can be removed from the original proto though (e.g. because the field is deprecated).

Reply:
This is all small potatoes though. Customers can always call `FooMethod(FooRequest const&)` in these cases.

I am fine not doing anything about this edge case.

Reply:
I agree that if we've never generated the "first match" overload (and therefore that it is a member of `omitted_rpcs`), that we could generate the second-match overload.

However, I would argue that we should still not generate the second-match overload, as you'd need to look at history (via the contents of the `emitted_rpcs`/`omitted_rpcs` lists) to be able to tell that this second match is really considered the first match.

That is, I'm claiming that we're all better off (customer included) if the "first match" is apparent directly from the proto.  Whether we generate that "real" first overload or not, then, is purely a question of backwards compatibility.

So, I vote for leaving things as they stand, no matter the "could"-ness.  Does that ring true to you?

Reply:
> I vote for leaving things as they stand.

Me too.

<hr />

(For the record. And I am probably mischaracterizing your argument)

I don't agree with the statement: "we should not generate [because we] need to look at ... `emitted_rpcs`/`omitted_rpcs`".

I do agree with the statement: "there is not enough value to justify doing the work.", which I think you are implying.

Reply:
> I don't agree with the statement: "we should not generate [because we] need to look at ... `emitted_rpcs`/`omitted_rpcs`".

Yes, I think that slightly mischaracterizes things, but we can chat later.

> I do agree with the statement: "there is not enough value to justify doing the work.", which I think you are implying.

No, it's not that I think there isn't enough value to do the work (the work would be easy, methinks), but rather that I think it would be a misstep.

So, I'll merge as-is.  I reckon that won't close any paths to "don't consider omitted signatures for first match", should we decide to go there in the future.